File: chat-demo-server/src/main/java/com/chat/challenge/dto/ChatMessageDTO.java
package com.chat.challenge.dto;

import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonInclude;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ChatMessageDTO {
	private ChatUserDTO sentByUser;	
	@NotNull
	private String content;
	private String createdAt;
}

File: chat-demo-client/src/app/models/chat-message.dto.ts
import { JsonObject, JsonProperty } from 'json2typescript';
import { ChatUserDTO } from './chat-user.dto';

@JsonObject('ChatMessageDTO')
export class ChatMessageDTO {
    @JsonProperty('sentByUser', ChatUserDTO)
    sentByUser: ChatUserDTO;
    @JsonProperty('content', String)
    content = '';
    @JsonProperty('createdAt', String)
    createdAt = '';
}
